gnark-prover/prover/marshal_non_inclusion.go

```diff
@@ -6,15 +6,18 @@ import (
 	"math/big"
 )
 
-type NonInclusionParametersJSON struct {
-	Roots          []string   `json:"roots"`
-	Values         []string   `json:"values"`
-	InPathIndices  []uint32   `json:"inPathIndices"`
-	InPathElements [][]string `json:"inPathElements"`
+type NonInclusionProofInputsJSON struct {
+	Root                 string   `json:"root"`
+	Value                string   `json:"value"`
+	PathIndex            uint32   `json:"pathIndex"`
+	PathElements         []string `json:"pathElements"`
+	LeafLowerRangeValue  string   `json:"leafLowerRangeValue"`
+	LeafHigherRangeValue string   `json:"leafHigherRangeValue"`
+	LeafIndex            uint32   `json:"leafIndex"`
+}
```

Comment:
shouldn't there only be one ProofInputsJSON marshall type now sine we only have one endpoint?

Reply:
Not necessarily, we compose combined circuit using decoupled inclusion/non-inclusion, I think this makes code easier to understand, less burden:

 https://github.com/Lightprotocol/light-protocol/pull/600/files#diff-8d6043aeb42b679af66348c1b85c2fbe129d39fba985e5859323b443e6f62ea2R9-R10